src/engraving/libmscore/segment.cpp

```diff
@@ -859,15 +859,15 @@ void Segment::sortStaves(QList<int>& dst)
     }
     for (EngravingItem* e : _annotations) {
         ElementType et = e->type();
-        if (!e->systemFlag()
-            || (et == ElementType::REHEARSAL_MARK)
-            || (et == ElementType::SYSTEM_TEXT)
-            || (et == ElementType::PLAYTECH_ANNOTATION)
-            || (et == ElementType::JUMP)
-            || (et == ElementType::MARKER)
-            || (et == ElementType::TEMPO_TEXT)
-            || (et == ElementType::VOLTA)
-            || (et == ElementType::TEXTLINE && e->systemFlag())) {
+        if (!e->systemFlag() || (e->isLinked()
```

Comment:
This code looks very complicated to read/debug. I would suggest simplifying it. The most complicated part of it can be simplified as follows:

static const std::set<ElementType> allowedTypes {
ElementType::REHEARSAL_MARK,
ElementType::PLAYTECH_ANNOTATION
...
}

if (!e->systemFlag() || (e->isLinked() && allowedTypes.contains(et)) {
...
}

This condition can probably be simplified even further

Reply:
Good idea. I've updated it with that suggestion. `set::contains()` is  c++20 it seems, but `::find()` works fine for that.